app/models/project_submission.rb

```diff
@@ -36,6 +36,10 @@ def unlike!(user = nil)
     self.liked = false
   end
 
+  def live_preview_url?
```

Comment:
It's a kind of hidden feature in Rails but the project submission model will already have an implicit `live_preview_url?` predicate defined because it can be null.

We can remove this new method and associated tests; and the code in the component should just work 🎉

Reply:
No problem! So I will just delete the method and tests, and check against `project_submission.live_preview_url.present?` in the component rendering conditions. Thanks for the tip! I wasn't sure if a new method should be added.